site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2193,6 +2204,62 @@ public function getGradedComponentsCountByGradingSections($g_id, $sections, $sec
         return $return;
     }
 
+
+    /**
+     * Get an array of Teams for a Gradeable matching the given rotating sections
+     *
+     * @param  string $g_id
+     * @param  array<int>  $sections
+     * @param  string $section_key
+     * @param  bool   $is_team
+     * @return array<int|string,int>
+     */
+    public function getVerifiedComponentsCountByGradingSections($g_id, $sections, $section_key, $is_team) {
+        $u_or_t = "u";
+        $users_or_teams = "users";
+        $user_or_team_id = "user_id";
+        if ($is_team) {
+            $u_or_t = "t";
+            $users_or_teams = "gradeable_teams";
+            $user_or_team_id = "team_id";
+        }
+        $return = [];
+        $params = [$g_id];
+        $where = "";
+        if (count($sections) > 0) {
+            $where = "WHERE active_version > 0 AND ({$section_key} IN " . $this->createParameterList(count($sections)) . ") IS NOT FALSE";
+            $params = array_merge($params, $sections);
+        }
+        $this->course_db->query(
+            "
+SELECT {$u_or_t}.{$section_key}, count({$u_or_t}.*) as cnt
+FROM {$users_or_teams} AS {$u_or_t}
+INNER JOIN (
+ SELECT * FROM gradeable_data AS gd
+ INNER JOIN (SELECT g_id, $user_or_team_id, max(active_version) as active_version FROM electronic_gradeable_version GROUP BY g_id, $user_or_team_id) AS egd on egd.g_id = gd.g_id AND egd.{$user_or_team_id} = gd.gd_{$user_or_team_id}
+ LEFT JOIN (
+ gradeable_component_data AS gcd
+ INNER JOIN gradeable_component AS gc ON gc.gc_id = gcd.gc_id AND gc.gc_is_peer = {$this->course_db->convertBoolean(false)}
+ )AS gcd ON gcd.gd_id = gd.gd_id WHERE gcd.g_id=? AND GCD_VERIFIER_ID IS NOT NULL
+) AS gd ON {$u_or_t}.{$user_or_team_id} = gd.gd_{$user_or_team_id}
+{$where}
+GROUP BY {$u_or_t}.{$section_key}
+ORDER BY {$u_or_t}.{$section_key}",
```

Comment:
It looks like you copied this code from elsewhere in this file.  Copying code is not conducive to future maintainability.  Is there a way for you to share more code with the original source?

Reply:
I referenced off of another code in the file because I saw that this code would work if I made a slight change to it to get the info I needed. I wasn't too sure how to query the database for the same info